### PR TITLE
Pythonic bitinformation

### PR DESCRIPTION
**Pull request by observingClouds**
This is work-in-progress for a pythonic way to retrieve the mutual information content.

### CODE REVIEW

**Comment by observingClouds** [comment]
This branch is now in a state where the basic functionality of `get_bitinformation` is ported to python and works in first tests reasonably well with Dask. Due to #98 and other issues with the wrapping of Julia code, large datasets could not be processed straight forward. Ideally, Julia and python would work better together in the future, but currently this seems to be a reasonable alternative.

However, the following features are still missing in the python implementation and should be added in a next step:
- enable masked values
- enable set_zero_insignificant (currently is set to False for the python implementation)
- any additional arguments to xb.get_bitinformation

**Review by aaronspring — APPROVED** [review]
Great work. Wasn’t expecting to see identical result from Julia and pythonic implementation. The main issue I see is that this means duplication of bitinformation.jl, I.e. each change there should be done here as well. On the other hand we could make julia and bitinformation.jl an optional install.

**Comment by observingClouds** [comment]
I second that the duplication is not ideal and I would like to avoid it, but currently it seems (at least to me) the easiest way to allow for larger datasets.

File: xbitinfo/xbitinfo.py [hunk 1]
```diff
@@ -106,12 +115,16 @@ def get_bitinformation(ds, dim=None, axis=None, label=None, overwrite=False, **k
       Label of the json to serialize bitinfo. When string, serialize results to disk into file ``{{label}}.json`` to be reused later. Defaults to ``None``.
     overwrite : bool
       If ``False``, try using serialized bitinfo based on label; if true or label does not exist, run bitinformation
+    implementation : str
+      Bitinformation algorithm implementation. Valid options are
```
[the remark this answers is not in the document]
Comment (observingClouds) [reply xbitinfo/xbitinfo.py:119]: Yeah, these are probably better options. I wanted to keep the reference to BitInformation.jl, but I think that can also be done in the docstring.

**Comment by aaronspring** [comment]
julia stays the default.

**Comment by observingClouds** [comment]
> julia stays the default.

Yes!!

**Pull request merged**